lib/puppet/transaction/event.rb

```diff
@@ -58,11 +60,13 @@ def to_data_hash
       'name' => @name,
       'status' => @status,
       'time' => @time.iso8601(9),
-      'redacted' => @redacted
+      'redacted' => @redacted,
+      'corrective_change' => @corrective_change,
     }
   end
 
   def property=(prop)
+    @property_instance = prop
```

Comment:
I spelunked back to 2009 on this (https://github.com/puppetlabs/puppet/commit/977595bd if you're curious) and I didn't find a reason why the property object itself wasn't cached. My *guess* is that it was following a pattern of "normalize things to strings" and there just hadn't been a reason before. But dunno.

That said, I'm not coming up with a reason adding this would cause problems.